packages/grid/x-data-grid/src/components/menu/columnMenu/GridColumnMenuContainer.tsx

```diff
@@ -9,7 +9,9 @@ import { GridColumnMenuContainerProps } from './GridColumnMenuProps';
 import { gridClasses } from '../../../constants/gridClasses';
 
 const StyledMenuList = styled(MenuList)(() => ({
-  minWidth: 248,
+  '@media (pointer: fine)': {
```

Comment:
I'm not sure that the constraint is on whether the primary pointer is fine (e.g. cursor) or not (e.g. touch screen). It feels like it's more a problem with screen width, so
```suggestion
  [theme.breakpoints.up('sm')]: {
```

Reply:
Makes sense, I think the same is done in MUI MenuList.